import-export-cli/cmd/exportAPIProduct.go

```diff
@@ -0,0 +1,174 @@
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/wso2/product-apim-tooling/import-export-cli/credentials"
+
+	"github.com/go-resty/resty"
+	"github.com/spf13/cobra"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+
+	"net/http"
+	"path/filepath"
+)
+
+var exportAPIProductName string
+var exportAPIProductVersion string
+var exportAPIProductProvider string
+var exportAPIProductFormat string
+var runningExportAPIProductCommand bool
+
+// ExportAPIProduct command related usage info
+const exportAPIProductCmdLiteral = "api-product"
+const exportAPIProductCmdShortDesc = "Export API Product"
+
+const exportAPIProductCmdLongDesc = "Export an API Product in an environment"
+
+const exportAPIProductCmdExamples = utils.ProjectName + ` ` + exportCmdLiteral + ` ` + exportAPIProductCmdLiteral + ` -n LeasingAPIProduct -e dev
+` + utils.ProjectName + ` ` + exportCmdLiteral + ` ` + exportAPIProductCmdLiteral + ` -n CreditAPIProduct -v 1.0.0 -r admin -e production
+NOTE: Both the flags (--name (-n) and --environment (-e)) are mandatory`
+
+// ExportAPIProductCmd represents the exportAPIProduct command
+var ExportAPIProductCmd = &cobra.Command{
+	Use: exportAPIProductCmdLiteral + " (--name <name-of-the-api-product> --provider <provider-of-the-api-product> --environment " +
+		"<environment-from-which-the-api-product-should-be-exported>)",
+	Short:   exportAPIProductCmdShortDesc,
+	Long:    exportAPIProductCmdLongDesc,
+	Example: exportAPIProductCmdExamples,
+	Run: func(cmd *cobra.Command, args []string) {
+		utils.Logln(utils.LogPrefixInfo + exportAPIProductCmdLiteral + " called")
+		var apiProductsExportDirectory = filepath.Join(utils.ExportDirectory, utils.ExportedApiProductsDirName)
+
+		cred, err := getCredentials(cmdExportEnvironment)
+		if err != nil {
+			utils.HandleErrorAndExit("Error getting credentials", err)
+		}
+
+		executeExportAPIProductCmd(cred, apiProductsExportDirectory)
+	},
+}
+
+func executeExportAPIProductCmd(credential credentials.Credential, exportDirectory string) {
+	runningExportAPIProductCommand = true
+	accessToken, preCommandErr := credentials.GetOAuthAccessToken(credential, cmdExportEnvironment)
+
+	if preCommandErr == nil {
+		adminEndpoint := utils.GetAdminEndpointOfEnv(cmdExportEnvironment, utils.MainConfigFilePath)
+		if exportAPIProductVersion == "" {
+			// If the user has not specified the version, use the version as 1.0.0
+			exportAPIProductVersion = utils.DefaultApiProductVersion
+		}
+		resp, err := getExportApiProductResponse(exportAPIProductName, exportAPIProductVersion, exportAPIProductProvider, exportAPIProductFormat, adminEndpoint,
+			accessToken)
+		if err != nil {
+			utils.HandleErrorAndExit("Error while exporting", err)
+		}
+		// Print info on response
+		utils.Logf(utils.LogPrefixInfo+"ResponseStatus: %v\n", resp.Status())
```

Comment:
Shall we add spaces before and after "+".
Saw this in few other places as well, shall we fix all the places?